cg/apps/sequencing_metrics_parser/api.py

```diff
@@ -1,28 +1,28 @@
+from datetime import datetime
 from pathlib import Path
 
-from cg.apps.sequencing_metrics_parser.parsers.bcl2fastq_to_sequencing_statistics import (
-    create_bcl2fastq_metrics,
-    create_bcl2fastq_undetermined_metrics,
-)
-from cg.apps.sequencing_metrics_parser.parsers.bcl_convert_to_sequencing_statistics import (
-    create_bcl_convert_undetermined_metrics,
-    create_sample_lane_sequencing_metrics_from_bcl_convert_metrics_for_flow_cell,
-)
-from cg.constants.demultiplexing import BclConverter
+from cg.apps.sequencing_metrics_parser.parsers.bcl_convert import BclConvertMetricsParser
+from cg.apps.sequencing_metrics_parser.parsers.utils import get_flow_cell_id
+from cg.constants.demultiplexing import UNDETERMINED
 from cg.models.flow_cell.flow_cell import FlowCellDirectoryData
 from cg.store.models import SampleLaneSequencingMetrics
 
 
 def create_sample_lane_sequencing_metrics_for_flow_cell(
-    flow_cell_directory: Path, bcl_converter: str
+    flow_cell_directory: Path,
 ) -> list[SampleLaneSequencingMetrics]:
-    """Parse the sequencing metrics data for the correct demultiplexing software
-    into the sequencing statistics model."""
-    if bcl_converter == BclConverter.BCLCONVERT:
-        return create_sample_lane_sequencing_metrics_from_bcl_convert_metrics_for_flow_cell(
-            flow_cell_directory
-        )
-    return create_bcl2fastq_metrics(flow_cell_directory)
+    """Parse the demultiplexing metrics data into the sequencing statistics model."""
+    metrics_parser = BclConvertMetricsParser(flow_cell_directory)
```

Comment:
I basically pasetd the content of `create_sample_lane_sequencing_metrics_from_bcl_convert_metrics_for_flow_cell` here